app/models/bgs_dependents/marriage.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+module BGSDependents
+  class Marriage < Base
+    attribute :ssn, String
+    attribute :first, String
+    attribute :middle, String
+    attribute :last, String
+    attribute :suffix, String
+    attribute :vet_ind, String
+    attribute :birth_date, String
+    attribute :alt_address, String
+    attribute :va_file_number, String
+    attribute :lives_with_vet, String
+    attribute :ever_married_ind, String
+    attribute :martl_status_type_cd, String
+
+    def initialize(dependents_application)
+      @dependents_application = dependents_application
+      @spouse_information = @dependents_application['spouse_information']
+
+      self.attributes = spouse_attributes
+    end
+
+    def format_info
+      attributes.with_indifferent_access
+    end
+
+    def alt_address
+      @dependents_application.dig('does_live_with_spouse', 'address')
+    end
+
+    def address(marriage_info)
+      dependent_address(
+        @dependents_application,
+        marriage_info['lives_with_vet'],
+        marriage_info['alt_address']
+      )
+    end
+
+    private
+
+    def spouse_attributes
+      marriage_info = {
+        'ssn': @spouse_information['ssn'],
+        'birth_date': @spouse_information['birth_date'],
+        'ever_married_ind': 'Y',
+        'martl_status_type_cd': marital_status,
+        'vet_ind': spouse_is_veteran,
+        'lives_with_vet': lives_with_vet,
+        'alt_address': alt_address
```

Comment:
`def alt_address` is defined as a public method on this class which overrides the attribute, so someone who calls `<an instance of BGSDependents::Marriage>.alt_address` will always step into the `def alt_address` logic of `@dependents_application.dig('does_live_with_spouse', 'address')` - is this intentional?

I think what you want here is to get rid of `def alt_address` and just initialize the attribute instead:
```ruby
'alt_address': @dependents_application.dig('does_live_with_spouse', 'address')
```
so that expensive `dig` logic only happens once in initialization.

Reply:
> def alt_address is defined as a public method on this class which overrides the attribute..

Sorry, this PR has some silly mistakes like that, I have/had tunnel vision 🤦 

> I think what you want here is to get rid of def alt_address and just initialize the attribute instead

I like that idea. I'm fairly certain all this `alt_address`/`lives_with_veteran` business was for the `address` in the end. I've just done address as an attribute like you suggested and placed the result of the alt/vet address in there bc I think that's the real goal with the lives w/ vet/alt address business. WDYT?